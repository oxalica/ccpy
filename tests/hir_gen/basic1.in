1 + 2
pass
(2, 3)
f((), x, -y.z, (None,))
__intrinsic__none
