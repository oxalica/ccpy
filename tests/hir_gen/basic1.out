Start(Local size = 8):
%0 = 1
%1 = 2
%2 = __intrinsic__get_global0()
%3 = "__builtin__add"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)
%0 = 2
%1 = 3
%0 = __intrinsic__tuple_make_(%0, %1)
%0 = "f"
%1 = __intrinsic__get_global0()
%2 = "__builtin__global_get"
%1 = __intrinsic__dict_get3(%1, %2, %3)
%0 = __intrinsic__v_call_(%1, %0)
%1 = __intrinsic__tuple_make_()
%2 = "x"
%3 = __intrinsic__get_global0()
%4 = "__builtin__global_get"
%3 = __intrinsic__dict_get3(%3, %4, %5)
%2 = __intrinsic__v_call_(%3, %2)
%3 = "y"
%4 = __intrinsic__get_global0()
%5 = "__builtin__global_get"
%4 = __intrinsic__dict_get3(%4, %5, %6)
%3 = __intrinsic__v_call_(%4, %3)
%4 = "z"
%5 = __intrinsic__get_global0()
%6 = "__builtin__getattr"
%5 = __intrinsic__dict_get3(%5, %6, %7)
%3 = __intrinsic__v_call_(%5, %3, %4)
%4 = __intrinsic__get_global0()
%5 = "__builtin__neg"
%4 = __intrinsic__dict_get3(%4, %5, %6)
%3 = __intrinsic__v_call_(%4, %3)
%4 = None
%4 = __intrinsic__tuple_make_(%4)
%1 = __intrinsic__tuple_make_(%1, %2, %3, %4)
%2 = __intrinsic__get_global0()
%3 = "__builtin__call"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)
%0 = "__intrinsic__wtf"
%1 = __intrinsic__get_global0()
%2 = "__builtin__global_get"
%1 = __intrinsic__dict_get3(%1, %2, %3)
%0 = __intrinsic__v_call_(%1, %0)
%1 = __intrinsic__tuple_make_()
%2 = __intrinsic__get_global0()
%3 = "__builtin__call"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)
%0 = 42
%0 = __intrinsic__id1(%0)
%0 = "__intrinsic__id1"
%1 = __intrinsic__get_global0()
%2 = "__builtin__global_get"
%1 = __intrinsic__dict_get3(%1, %2, %3)
%0 = __intrinsic__v_call_(%1, %0)
%0 = "f"
%1 = __intrinsic__get_global0()
%2 = "__builtin__global_get"
%1 = __intrinsic__dict_get3(%1, %2, %3)
%0 = __intrinsic__v_call_(%1, %0)
%1 = 1
%1 = __intrinsic__tuple_make_(%1)
%2 = __intrinsic__get_global0()
%3 = "__builtin__call"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)
%1 = 2
%1 = __intrinsic__tuple_make_(%1)
%2 = __intrinsic__get_global0()
%3 = "__builtin__call"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)
%1 = 3
%1 = __intrinsic__tuple_make_(%1)
%2 = __intrinsic__get_global0()
%3 = "__builtin__call"
%2 = __intrinsic__dict_get3(%2, %3, %4)
%0 = __intrinsic__v_call_(%2, %0, %1)


